docs/doc2.md

```diff
@@ -77,17 +77,22 @@ The OpCodes are split into two categories: *"this spend is only valid if X"* and
-* **AGG_SIG_ME - [57] - (57 0xpubkey 0xmessage)**: The spend is only valid if the attached aggregated signature contains a signature from the specified public key of that message concatenated with the coin's id.
-* **ASSERT_FEE - [58] - (58 amount)**: The spend is only valid if there is unused value in this transaction equal to *amount*, which is explicitly to be used as the fee.
-* **ASSERT_TIME_EXCEEDS - [59] - (59 time)**: The spend is only valid if the given time has passed.
-* **CREATE_ANNOUNCEMENT - [60] - (60 message)**: This creates an ephemeral announcement with an ID dependent on the coin that creates it. Other coins can then assert an announcement exists for inter-coin communication inside a block.
+* **AGG_SIG_UNSAFE - [49] - (49 0xpubkey 0xmessage)**: This spend is only valid if the attached aggregated signature contains a signature from the given public key of the given message. This is labeled unsafe because if you sign a message once, any other coins you have that require that signature may potentially also be unlocked. It's probably better just to use AGG_SIG_ME because of the natural entropy introduced by the coin ID. 
+* **AGG_SIG_ME - [50] - (50 0xpubkey 0xmessage)**: This spend is only valid if the attached aggregated signature contains a signature from the specified public key of that message concatenated with the coin's ID.
+* **CREATE_COIN - [51] - (51 0xpuzzlehash amount)**: If this spend is valid, then create a new coin with the given puzzlehash and amount.
+* **ASSERT_FEE - [52] - (52 amount)**: This spend is only valid if there is unused value in this transaction equal to *amount*, which is explicitly to be used as the fee.
+* **CREATE_COIN_ANNOUNCEMENT - [60] - (60 message)**: If this spend is valid, this creates an ephemeral announcement with an ID dependent on the coin that creates it. Other coins can then assert an announcement exists for inter-coin communication inside a block.
+* **ASSERT_COIN_ANNOUNCEMENT - [61] - (61 0xannouncementID)**: This spend is only valid if there was an announcement in this block matching the announcementID.  The announcementID is the message that was announced concatenated with the coin ID of the coin that announced it `announcementID == (message + coinID)`.
+* **CREATE_PUZZLE_ANNOUNCEMENT - [62] - (62 message)**: If this spend is valid, this creates an ephemeral announcement with an ID dependent on the puzzle that creates it. Other coins can then assert an announcement exists for inter-coin communication inside a block.
+* **ASSERT_PUZZLE_ANNOUNCEMENT - [63] - (63 0xannouncementID)**: This spend is only valid if there was an announcement in this block matching the announcementID.  The announcementID is the message that was announced concatenated with the puzzle hash of the coin that announced it `announcementID == (message + puzzle_hash)`.
```

Comment:
Should clarify that its a hash of those things concatenated

Reply:
the order is `puzzle_hash + message`

Reply:
Oh yes, do you think it should be Chialisp like `(sha256 coinID message)` or just some pseudo code like `hash(coinID + message)`